hilti/runtime/include/types/struct.h

```diff
@@ -27,8 +27,23 @@ inline auto& value_or_exception(const std::optional<T>& t, const char* location)
 } // namespace struct_
 
 namespace detail::adl {
+
+template<typename>
+constexpr std::false_type has__str__helper(long);
+
+template<typename T>
+constexpr auto has__str__helper(int) -> decltype(std::declval<T>().__str__(), std::true_type{});
+
+template<typename T>
+using has__str__ = decltype(has__str__helper<T>(0));
+
 template<typename T, typename std::enable_if_t<std::is_base_of<trait::isStruct, T>::value>* = nullptr>
 inline std::string to_string(const T& x, adl::tag /*unused*/) {
+    if constexpr ( has__str__<T>() ) {
```

Comment:
Can we add a test for this to the [to_string testsuite](hilti/runtime/src/tests/to_string.cc)? This should not be hard.

Reply:
Added to ``struct`` tests, that's where the current "to_string" test resides.